### PR TITLE
Docs: add more examples for `no-sequences` rule

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[X] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Add more examples.

Fixes https://github.com/eslint/eslint/issues/14572

#### Is there anything you'd like reviewers to focus on?

Nope

### CODE REVIEW

**Review by aladdin-add — APPROVED** [review]
LGTM, thanks!

**Review by mdjermanovic — COMMENTED** [review]
The added examples LGTM, but since we already had a similar example with arrow function (right above the new examples) I think the idea in https://github.com/eslint/eslint/issues/14572#issuecomment-834897231 was to add a note. @nzakas ?

**Comment by nzakas** [comment]
Yes, I think we should add a note that explicitly states this exception case as it’s different from anywhere else in the rule.

**Comment by nzakas** [comment]
@snitin315 are you still working on this?

**Comment by snitin315** [comment]
Yes, thanks for the ping. I will push an update today.

**Comment by snitin315** [comment]
Done ✅

**Review by nzakas — COMMENTED** [review]
I was more thinking we should have a specific paragraph in the page saying something like:

### Note about arrow function bodies

If an arrow function body is a statement rather than a block, and that statement contains a sequence, you need to use double parentheses around the statement to indicate that the sequence is intentional.

(include examples here)

**Review by nzakas — APPROVED** [review]
Exactly what I was looking for. Thank you!

File: docs/rules/no-sequences.md [hunk 1]
```diff
@@ -63,8 +61,32 @@ switch ((val = foo(), val)) {}
...
+
+If an arrow function body is a statement rather than a block, and that statement contains a sequence, you need to use double parentheses around the statement to indicate that the sequence is intentional.
+
+Examples of **incorrect** code for arrow functions:
+
+```js
+/*eslint no-sequences: "error"*/
+const foo = (val) => (console.log('bar'), val);
+
+const foo = () => ((bar = 123), 10));
```
Comment (fisker) [inline docs/rules/no-sequences.md:76]: These is an extra `)` in this example.

Comment (snitin315) [reply docs/rules/no-sequences.md:76]: Good catch, thanks. I will send a fix.

**Pull request merged**